service/tbc/tbc.go

```diff
@@ -638,21 +631,15 @@ func (s *Server) peerConnect(ctx context.Context, peerC chan string, p *peer) {
 			return
 		}
 	}
-	if len(bhs) != 1 {
-		// XXX fix multiple tips
-		panic(len(bhs))
-	}
-	log.Debugf("block header best hash: %s", bhs[0].Hash)
+	log.Debugf("block header best hash: %s", bhb.Hash)
 
-	err = s.getHeaders(ctx, p, bhs[0].Header)
+	err = s.getHeaders(ctx, p, bhb.Header)
 	if err != nil {
```

Comment:
I think this error assignment and check could be nested:
```suggestion
	if err = s.getHeaders(ctx, p, bhb.Header); err != nil {
```